nrf52-hal-common/src/spim.rs

```diff
@@ -375,6 +354,305 @@ where
+    ///
+    /// This is an async polling version of `transfer()`.  You need to call
+    /// `.poll_complete()` on the returned object until it returns `true`.  A
+    /// good time to do that would be after receiving a SPI interrupt, for
+    /// example.
+    ///
+    /// This method implements a complete read transaction, which consists of
+    /// the master transmitting what it wishes to read, and the slave responding
+    /// with the requested data.
+    ///
+    /// Uses the provided chip select pin to initiate the transaction. Transmits
+    /// all bytes in `buffer`, then receives an equal number of bytes.
+    pub fn transfer_polling<'b>(
+        &'b mut self,
+        buffer: &'b mut [u8],
+    ) -> Result<SpiTransfer<'b, T>, Error> {
+        SpiTransfer::new(self.spim, buffer)
+    }
+
+    /// Read and write from a SPI slave, using separate read and write buffers
+    ///
+    /// This is an async polling version of `transfer_split_even()`.  You need to
+    /// call `.poll_complete()` on the returned object until it returns `true`.
+    /// A good time to do that would be after receiving a SPI interrupt, for
+    /// example.
+    ///
+    /// This method implements a complete read transaction, which consists of
+    /// the master transmitting what it wishes to read, and the slave responding
+    /// with the requested data.
+    ///
+    /// Uses the provided chip select pin to initiate the transaction. Transmits
+    /// all bytes in `tx_buffer`, then receives bytes until `rx_buffer` is full.
+    ///
+    /// If `tx_buffer.len() != rx_buffer.len()`, the transaction will stop at the
+    /// smaller of either buffer.
+    pub fn transfer_split_even_polling<'b>(
+        &'b mut self,
+        tx_buffer: &'b [u8],
+        rx_buffer: &'b mut [u8],
+    ) -> Result<SpiEvenTransfer<'b, T>, Error> {
+        SpiEvenTransfer::new(self.spim, tx_buffer, rx_buffer)
+    }
+
+    /// Read and write from a SPI slave, using separate read and write buffers
+    ///
+    /// This is an async polling version of `transfer_split_uneven()`.  You need to
+    /// call `.poll_complete()` on the returned object until it returns `true`.
+    /// A good time to do that would be after receiving a SPI interrupt, for
+    /// example.
+    ///
+    /// This method implements a complete read transaction, which consists of
+    /// the master transmitting what it wishes to read, and the slave responding
+    /// with the requested data.
+    ///
+    /// Uses the provided chip select pin to initiate the transaction. Transmits
+    /// all bytes in `tx_buffer`, then receives bytes until `rx_buffer` is full.
+    ///
+    /// This method is more complicated than the other `transfer` methods because
+    /// it is allowed to perform transactions where `tx_buffer.len() != rx_buffer.len()`.
+    /// If this occurs, extra incoming bytes will be discarded, OR extra outgoing bytes
+    /// will be filled with the `orc` value.
```

Comment:
What's the `orc` value?

Reply:
This is copy pasted from the previous code, it's a constructor argument given when creating the SPI master.  I can refer to the constructor here if that helps.

Reply:
Looks good, thanks!